### PR TITLE
BTC wallet update

### PR DESCRIPTION
**Pull request by alex-miller-0**
* Enables `bech32` and `legacy` wallets
* Fixes tx history issues

### CODE REVIEW

File: src/sdk/sdkSession.ts [hunk 1]
```diff
@@ -564,46 +564,66 @@ class SDKSession {
     const processedTxs: any[] = [];
     const txs = JSON.parse(JSON.stringify(this.btcTxs));
     txs.forEach((tx) => {
-      tx.recipient = tx.outputs[0].addr;
-      tx.incoming = allAddrs.indexOf(tx.recipient) > -1;
+      // Determine if this is an outgoing transaction or not based on inputs.
+      // If *any* input comes from one of our addresses, we consider the 
+      // transaction to be "outgoin".
```
Comment (douglance) [inline src/sdk/sdkSession.ts:569]:
```suggestion
      // transaction to be "outgoing".
```

File: src/sdk/sdkSession.ts [hunk 2]
```diff
@@ -564,46 +564,66 @@ class SDKSession {
...
+      let outgoing = false;
+      tx.inputs.forEach((input) => {
+        if (allAddrs.indexOf(input.addr) > -1) {
+          outgoing = true;
+        }
+      })
+      tx.incoming = !outgoing;
```
Comment (douglance) [inline src/sdk/sdkSession.ts:576]: I don't think it's worth holding up the PR and retesting everything for this, but this type of logic can also be written like this: 

```ts
tx.incoming = tx.inputs.every(input => allAddrs.indexOf(input.addr) > -1)
```

which could be a bit easier to parse for some people.

File: src/util/helpers.ts [hunk 3]
```diff
@@ -272,26 +274,38 @@ function _fetchBtcTxs(addresses, cb, txs=[], offset=0, isFirstCall=true) {
...
+            // Only add the transaction if its hash is not already in the array.
+            // NOTE: There may be an edge case. I noticed in one case we got
+            // a result saying `vout_sz=2` but which only had one output in its array...
+            let shouldInclude = true;
+            txs.forEach((_tx) => {
+                if (_tx.id === ftx.id) {
+                    shouldInclude = false;
+                }
+            })
+            if (shouldInclude) {
```
Comment (douglance) [inline src/util/helpers.ts:287]: Same as above. This could be: 

```ts
txs.every(_tx => _tx.id !== ftx.id)
```

**Review by douglance — APPROVED** [review]

**Comment by douglance** [comment]
I don't really have a way to test it< so I just marked a couple of stylistic things that are probably just my preference.

File: src/util/helpers.ts (hunk 3 above)
Comment (alex-miller-0) [reply src/util/helpers.ts:282]: [2211973](https://github.com/GridPlus/lattice-manager/pull/219/commits/221197375f71b9b0441d0868439625d6cc3ddddb)

File: src/sdk/sdkSession.ts (hunk 2 above)
Comment (alex-miller-0) [reply src/sdk/sdkSession.ts]: [2211973](https://github.com/GridPlus/lattice-manager/pull/219/commits/221197375f71b9b0441d0868439625d6cc3ddddb)

File: src/sdk/sdkSession.ts (hunk 1 above)
Comment (alex-miller-0) [reply src/sdk/sdkSession.ts]: [2211973](https://github.com/GridPlus/lattice-manager/pull/219/commits/221197375f71b9b0441d0868439625d6cc3ddddb)

**Pull request merged**